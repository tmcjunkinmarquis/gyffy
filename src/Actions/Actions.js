export const inputLocation = (location)=>({
  type: 'INPUT_LOCATION',
  location
});

export const loadStations = (stations)=>({
  type: 'LOAD_STATIONS',
  stations
});