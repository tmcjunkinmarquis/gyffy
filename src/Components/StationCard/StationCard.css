.fuel-station {
  text-align: center;
  height: 200px;
  width: 150px;
  box-shadow: 1px 1px 3px #6A9ECF;
  background-color:white;
  margin: 10px;
};


.name {
  font-weight: 100;
  text-shadow: none;
  color: darkslategray;
  
}

.address {
  text-shadow: none;
  color: darkslateblue;
  
}

.fuel-type {
  color: hotpink;
}

.connector-type {
  font-size: smaller;
}