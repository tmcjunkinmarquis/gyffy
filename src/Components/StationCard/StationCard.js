import React, { Component } from 'react';
import './StationCard.css';
import PropTypes from 'prop-types';
import { setId, setSelectedStation } from '../../Actions/Actions';
import { connect } from 'react-redux';

export class Station extends Component {
 
  handleClick = ()=>{
    this.props.setId(this.props.station.id);
    this.props.setSelectedStation(this.props.station);
  }
  
  showConnectortype = ()=>{
    return this.props.station.connectorTypes.map((type)=>{
      return <span 
        className="connector-type"
        key={type}>
        {`${type}`}</span>;
    });
  };

  render() { 
    return (
      <div
        onClick={this.handleClick}
        className="fuel-station">
        <p className="name">{this.props.station.name}</p>
        <p className="address">{this.props.station.address}</p>
        <p className="fuel-type fuel-type">{this.props.station.fuelType}</p>
        {this.props.station.fuelType === 'ELEC' ? <span><p>connectors: </p>{this.showConnectortype()}</span> : ''}
      </div>
    );
  } 
};

export const mapDispatchToProps = (dispatch)=>({
  setId: (id)=>dispatch(setId(id)),
  setSelectedStation: (station) => dispatch(setSelectedStation(station))
});

export default connect(null, mapDispatchToProps)(Station);

Station.propTypes = {
  station: PropTypes.object,
  setId: PropTypes.func,
  setSelectedStation: PropTypes.func
};