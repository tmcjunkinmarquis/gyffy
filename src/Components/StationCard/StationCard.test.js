import React from 'react';
import { shallow } from 'enzyme';
import { Station, mapDispatchToProps } from './StationCard';
import { setId } from '../../Actions/Actions';

describe('StationCard', () => {
  
  describe('Station', ()=>{
    let wrapper;
    let mockSetId;
    let mockSetSelectedStation;
    let mockProps;

    beforeEach(()=>{
      mockSetId = jest.fn();
      mockSetSelectedStation = jest.fn();
      mockProps = {
        station:{name: 'Bubbas',
          address: '123 right here',
          fuelType: 'ELEC',
          connectorTypes: [''],
          id: 3}
      };
      wrapper = shallow(
        <Station 
          setId = {mockSetId}
          setSelectedStation = {mockSetSelectedStation}
          {...mockProps}
        />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    let wrapper;
    let mockSetId;
    let mockSetSelectedStation;
    let mockProps;

    beforeEach(()=>{
      mockProps = {
        station: {
          name: 'Bubbas',
          address: '123 right here',
          fuelType: 'ELEC',
          connectorTypes: [''],
          id: 3
        }
      };
      wrapper = shallow(
        <Station
          setId={mockSetId}
          setSelectedStation={mockSetSelectedStation}
          
          {...mockProps}
        />);

      mockSetId = jest.fn();
      mockSetSelectedStation = jest.fn();
      
    });
    
    it('calls dispatch with a setId action when handleClick is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setId(5); 
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setId(5);
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });  
  }); 
});
