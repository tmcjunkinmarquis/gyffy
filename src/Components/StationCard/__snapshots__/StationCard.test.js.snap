// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationCard Station should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Station
    setId={[Function]}
    setSelectedStation={[Function]}
    station={
        Object {
            "address": "123 right here",
            "connectorTypes": Array [
              "",
            ],
            "fuelType": "ELEC",
            "id": 3,
            "name": "Bubbas",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="name"
>
          Bubbas
</p>,
        <p
          className="address"
>
          123 right here
</p>,
        <p
          className="fuel-type fuel-type"
>
          ELEC
</p>,
        <span>
          <p>
                    connectors: 
          </p>
          <span
                    className="connector-type"
          />
</span>,
      ],
      "className": "fuel-station",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bubbas",
          "className": "name",
        },
        "ref": null,
        "rendered": "Bubbas",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "123 right here",
          "className": "address",
        },
        "ref": null,
        "rendered": "123 right here",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ELEC",
          "className": "fuel-type fuel-type",
        },
        "ref": null,
        "rendered": "ELEC",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              connectors: 
</p>,
            Array [
              <span
                className="connector-type"
/>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "connectors: ",
            },
            "ref": null,
            "rendered": "connectors: ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "",
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "connector-type",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="name"
>
            Bubbas
</p>,
          <p
            className="address"
>
            123 right here
</p>,
          <p
            className="fuel-type fuel-type"
>
            ELEC
</p>,
          <span>
            <p>
                        connectors: 
            </p>
            <span
                        className="connector-type"
            />
</span>,
        ],
        "className": "fuel-station",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bubbas",
            "className": "name",
          },
          "ref": null,
          "rendered": "Bubbas",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "123 right here",
            "className": "address",
          },
          "ref": null,
          "rendered": "123 right here",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ELEC",
            "className": "fuel-type fuel-type",
          },
          "ref": null,
          "rendered": "ELEC",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                connectors: 
</p>,
              Array [
                <span
                  className="connector-type"
/>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "connectors: ",
              },
              "ref": null,
              "rendered": "connectors: ",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "",
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "connector-type",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
